encoding/ssz/helpers.go

```diff
@@ -113,6 +114,51 @@ func Pack(serializedItems [][]byte) ([][]byte, error) {
 	return chunks, nil
 }
 
+// PackByChunk a given byte array's final chunk with zeroes if needed.
+func PackByChunk(serializedItems [][]byte) ([][bytesPerChunk]byte, error) {
+	areAllEmpty := true
+	for _, item := range serializedItems {
+		if !bytes.Equal(item, []byte{}) {
+			areAllEmpty = false
+			break
+		}
+	}
+	// If there are no items, we return an empty chunk.
+	if len(serializedItems) == 0 || areAllEmpty {
+		emptyChunk := [bytesPerChunk]byte{}
+		return [][bytesPerChunk]byte{emptyChunk}, nil
+	} else if len(serializedItems[0]) == bytesPerChunk {
+		// If each item has exactly BYTES_PER_CHUNK length, we return the list of serialized items.
+		chunks := make([][bytesPerChunk]byte, 0, len(serializedItems))
+		for _, c := range serializedItems {
+			chunks = append(chunks, bytesutil.ToBytes32(c))
+		}
+		return chunks, nil
+	}
+	// We flatten the list in order to pack its items into byte chunks correctly.
+	var orderedItems []byte
+	for _, item := range serializedItems {
+		orderedItems = append(orderedItems, item...)
+	}
```

Comment:
instead of doing the extra loop at the top of the function to check if `areAllEmpty`, could you let that condition get to right after this loop and then add this check?
```
if len(orderedItems) == 0 {
    return [][bytesPerChunk]byte{emptyChunk}, nil
}
```

Reply:
example showing that a flattened 2d array consisting of all empty sub-arrays would be len = 0: 
https://play.golang.org/p/onJkAnvnqfj